app/helpers/timelines_helper.rb

```diff
@@ -182,4 +182,37 @@ def push_timeline_options(timeline, target = gon)
   def timeline_options
     OpenStruct.new @timeline.options
   end
+
+  def new_timeline_link(project, &_block)
+    link_to({ controller: '/timelines', action: 'new', project_id: project },
+            title: l('timelines.new_timeline'),
+            class: 'button -alt-highlight') do
+      yield if block_given?
+    end
```

Comment:
If you want to use the `&block` variable, you should be able to pass it directly into the `link_to` method which would remove a stackframe: `link_to('foo', 'bar', &block)`